slides/week0/week0.org

```diff
@@ -0,0 +1,484 @@
+- our code is in {}
+- *cout* tells the computer to write the letters to the screen
+- *<<* tells the computer what to write out
+- line 1: inclues a header file (covered later)
+- line 4: defines the main function
+- line 6: tells the computer that the program ran successfully
+#+BEGIN_SRC C++ -n :exports both :includes <iostream> :defines cout std::cout
+  #include <iostream>
+  using namespace std;
+
+  int main() {
+    cout << "Hello World!";
+    return 0;
+  }
+#+END_SRC
+
+* Most useful program ever
+#+BEGIN_NOTES
+This should naturally lead into why variables are useful. Without them
+all programs would be deterministic.
+#+END_NOTES
+#+BEGIN_SRC C++ -n :exports both :includes <iostream> :defines cout std::cout
+  int main() {
+    cout << 2 + 2;
+    return 0;
+  }
+#+END_SRC
+
+* Variables
+- Alias for a value
+- the value can be changed without changing the alias
+
+* Primitive Variables Types
+| Name | Description |
+|-------+------|
+| =bool= | Can be either *true* or *false*|
+| =char= | a character like 'c' or '+'|
+| =int= | A whole number *bounded|
+| =double= | a decimal number *bounded|
+- What makes a variable primitive?
+
+* Primitive Variables?
+[[file:https://i.imgur.com/Wo0ovX5.jpg]]
+- But Actually... It means that it cannot be divided into smaller parts
+- A number is just a number
+
+* Important Symbols
+#+BEGIN_NOTES
+No lengthy explanation of << and stdout
+#+END_NOTES
+| symbol | meaning |
+|-------+------|
+| =//= | Signifies a comment, The computer will not try to compile that line |
+| === | assignment operator, value of right is assigned to variable on left |
+| =<<= | sends something to the OS which displays it in the terminal |
+
+* Math Operators
+- order of operations still applies
+- ~()~ > ~*,/~ > ~+,-~
+| symbol | meaning |
+|-------+------|
+| =()= | =groups a statement= |
+| =*= | =multiplies two numbers= |
+| =/= | =divides a number by another= |
+| =+= | =adds two numbers together= |
+| =-= | =subtracts two numbers= |
+
+* Variable Syntax
+| =type= | =name= | === | =value= | ; |
+|-------+------|
+| =bool= | =var= | = | true | ; |
+| =char= | =character= | = | 'c' | ; |
+| =int= | =intVar= | = | 2 | ; |
+| =double= | =double_var= | = | 2.2647 | ; |
+
+* Variables
+- =my_var= now represents the value 2
+#+BEGIN_SRC C++ -n :exports both :includes <iostream> :defines cout std::cout
+  int main() {
+    int my_var = 2;
+    cout << my_var + my_var;
+    return 0;
+  }
+#+END_SRC
+
+* Declaration vs Definition
+- int num = 1;
+    - creates the variable num of type int and assigns it the value 1
+- num = 4;
+    - the type is already known from above so this line just assigns it the value 4
+
+* Variables
+- Variables can have their assigned value changed on the fly
+#+BEGIN_SRC C++ -n :exports both :includes <iostream> :defines cout std::cout
+  int main() {
+    int my_var = 2;
+    cout << my_var + my_var;
+    my_var = 4;
+    cout << my_var + my_var;
+    return 0;
+  }
+#+END_SRC
+
+* Questions??
```

Comment:
Personally I prefer just one question mark but I won't be offended if you ignore this comment 😄